xmlrunner/result.py

```diff
@@ -156,8 +156,10 @@ def __init__(self, test_result, test_method, outcome=SUCCESS, err=None, subTest=
 
         self.test_name = testcase_name(test_method)
         self.test_id = test_method.id()
+        self.subDescription = None
```

Comment:
should the `addSubtest` method also be updated to handled cases where the the subtest succeeded. This can be detected when no err is provided to the `addSubtest` method.

Reply:
From the docs for `addSubTest(test, subtest, outcome)`: 

>  If outcome is None, the subtest succeeded

Link: https://docs.python.org/3.4/library/unittest.html?highlight=unittest#unittest.TestResult.addSubTest

Reply:
I don't know but with my changes the successful subtest does not get reported as failed or anything (or so I think)

Reply:
My question was whether this method should also be updated to include successful subtests in the result XML. The docs indicate that unittest does nothing by default, but I think that the xml runner might want to do something different in this case. 

Since subtest successes aren't currently tracked right now, the reports generate variable number of tests executed based on subtest success / failure. Additionally, as is, it is not possible to track the history of subtests since all the successes are seen as a no-run for the subtest.

Reply:
The reasoning I was thinking through was the second paragraph above, though I may not have articulated it well. The test history, reliability rate, and time taken for sub tests are less meaningful unless you include successes as part of the report. A sub test might typically fail in 3 ms but take 32 seconds when succeeding - not recording the successes may mask test infrastructure issues and makes test analysis more difficult. 

Another way to look at the above is that a subtest success should be treated like a test success. Based on the current implementation, test successes are considered interesting and hence included in the final report. Why would subtest successes be different? unitest does nothing with `addSuccess` as well, but the library overloads this functionality to add the result to the xml.

I'm not sure I understand your point on confusing others, could you elaborate more on the point? I'm also not sure how generators making subtests easy is relevant to this confusion. Agreed that the use of the flag would simplify integration and avoid back-compat problems / confusion. 

I can open an issue to get feedback from others if that would be more useful. Though based on my understanding of how subtests are used within unittest, I'd expect the framework to treat the results the same as a regular test.